hadoop-hdds/interface-client/src/main/proto/hdds.proto

```diff
@@ -307,6 +307,14 @@ enum ReplicationFactor {
     ZERO = 0; // Invalid Factor
 }
 
+enum StorageType {
+    DISK_TYPE = 1;
+    SSD_TYPE = 2;
+    ARCHIVE_TYPE = 3;
+    RAM_DISK_TYPE = 4;
+    PROVIDED_TYPE = 5;
```

Comment:
I don't recall who, but back in HDFS land after storage policy feature was implemented, someone (Arpit or Anu?) regrettably said it was a mistake to have these types continuous, because it wasn't possible to add any storage types in between. It might be a good idea to space them out.

@szetszwo do you recall?

Reply:
I also don't recall.

> ... it was impossible to insert any new storage types in between. ...
Not really -- we may use 1.1 if there is a need.

Also, why it has to insert in between?  We may put the new policy at the end.  I don't think Storage policy has a [total ordering](https://en.wikipedia.org/wiki/Total_order).